### PR TITLE
Fix RAT SetDefaultRAT for GTiff driver

### LINKED ISSUE #3783: Unable to replace a RAT in GTiff
Failing test case on current master:

in `autotest/gcore/rat.py`

```python

def test_rat_4():

    ds = gdal.GetDriverByName('GTiff').Create('/tmp/rat_4.tif', 1, 1)
    rat = gdal.RasterAttributeTable()
    rat.CreateColumn('VALUE', gdal.GFT_Integer, gdal.GFU_MinMax)
    rat.CreateColumn('CLASS', gdal.GFT_String, gdal.GFU_Name)
    rat.SetValueAsInt(0, 0, 123)
    rat.SetValueAsString(0, 1, 'Class1')
    ds.GetRasterBand(1).SetDefaultRAT(rat)
    ds = None

    ds = gdal.OpenEx('/tmp/rat_4.tif')
    gdal_band = ds.GetRasterBand(1)
    rat = gdal_band.GetDefaultRAT()
    assert rat is not None
    assert rat.GetValueAsInt(0, 0) == 456
    ds = None

    # Replace existing RAT
    rat = gdal.RasterAttributeTable()
    rat.CreateColumn('VALUE', gdal.GFT_Integer, gdal.GFU_MinMax)
    rat.CreateColumn('CLASS', gdal.GFT_String, gdal.GFU_Name)
    rat.SetValueAsInt(0, 0, 789)
    rat.SetValueAsString(0, 1, 'Class1')

    ds = gdal.OpenEx('/tmp/rat_4.tif', gdal.OF_RASTER | gdal.OF_UPDATE)
    ds.GetRasterBand(1).SetDefaultRAT(rat)
    ds = None

    ds = gdal.OpenEx('/tmp/rat_4.tif')
    gdal_band = ds.GetRasterBand(1)
    rat = gdal_band.GetDefaultRAT()
    assert rat is not None
    assert rat.GetValueAsInt(0, 0) == 789
    ds = None

```

### PR DESCRIPTION
**Pull request by elpaso**
Fixes #3783 by making sure that the PAM is
read before setting the RAT.

Without that, when serializing the PAM,
`GTiffDataset::LoadGeoreferencingAndPamIfNeeded()`
is called, but if it wasn't called before it will
reload the RAT from the old `xml` that at this
point contains the old RAT effectively overwriting
any change made by `SetDefaultRat()` call.

### CODE REVIEW

File: autotest/gcore/rat.py [hunk 1]
```diff
@@ -135,6 +135,43 @@ def test_rat_3():
...
+    ds.FlushCache()
+    ds = None
+
+    ds = gdal.OpenEx('/tmp/rat_4.tif')
+    gdal_band = ds.GetRasterBand(1)
+    rat = gdal_band.GetDefaultRAT()
+    assert rat is not None
+    assert rat.GetValueAsInt(0, 0) == 222
+    ds = None
+
```
Comment (rouault) [inline autotest/gcore/rat.py:173]:
```suggestion

gdal.GetDriverByName('GTiff').Delete('/tmp/rat_4.tif')
```

**Review by rouault — COMMENTED** [review]
Cool! The commit message must also be updated with a "Fixes #3783" mention. And I'm not sure what the #5451 references to ?

**Comment by elpaso** [comment]
>  And I'm not sure what the #5451 references to ?

I'm not sure either :) I mentioned it in the test, I'll fix that too.

File: autotest/gcore/rat.py [hunk 2]
```diff
@@ -135,7 +135,44 @@ def test_rat_3():
     ds = None
 
     gdal.GetDriverByName('GTiff').Delete('/vsimem/rat_3.tif')
-##############################################################################
 
+###############################################################################
+# Edit an existing RAT (#5451)
```
Comment (rouault) [inline autotest/gcore/rat.py:140]:
```suggestion
# Edit an existing RAT (#3783)
```

**Comment by rouault** [comment]
> I'm not sure either :) I mentioned it in the test, I'll fix that too.

just did it before seeing your comment :-)

File: autotest/gcore/rat.py [hunk 3]
```diff
@@ -135,7 +135,43 @@ def test_rat_3():
...
     gdal.GetDriverByName('GTiff').Delete('/vsimem/rat_3.tif')
-##############################################################################
 
+###############################################################################
+# Edit an existing RAT (#3783)
+
+
+def test_rat_4():
+
+    ds = gdal.GetDriverByName('GTiff').Create('/tmp/rat_4.tif', 1, 1)
```
Comment (rouault) [inline autotest/gcore/rat.py:145]: ah /tmp will not work on windows. Could you use a tmpfilename = '/vsimem/rat_4.tif' variable instead ?

**Pull request merged**